makefiles/messages.mk

```
@@ -56,17 +56,16 @@ else
 endif
 
 	ECHO_PREPROCESSING = $(ECHO_BEGIN)$(PRINT_FORMAT_RED) "Preprocessing $<"$(ECHO_END); $(ECHO_PIPEFAIL) (
+	ECHO_PREPROCESSING_XSWIFT = $(ECHO_BEGIN)$(PRINT_FORMAT_RED) "Preprocessing $(subst $(_SPACE),$(_COMMA) ,$^)"$(ECHO_END); $(ECHO_PIPEFAIL) (
```

We may be able to make use of `$(foreach` (assuming `$^` is a space-separated list) to avoid having to define `_SPACE` and `_COMMA`.
```suggestion
	ECHO_PREPROCESSING_XSWIFT = $(ECHO_BEGIN)$(PRINT_FORMAT_RED) "Preprocessing $(foreach x,$^,$(x),)"$(ECHO_END); $(ECHO_PIPEFAIL) (
```

---

This would add a trailing comma after the last filename, wouldn't it?